### PR TITLE
VER-82167: Fix most unit tests

### PR DESCRIPTION
**Pull request by jonathanl-bq**
This PR fixes most unit tests. There are still a few that need to be updated, but this fixes the rest of the tests. Integration tests and other suites will be covered in later PRs.

### CODE REVIEW

File: packages/vertica-nodejs/lib/client.js [hunk 1]
```diff
@@ -238,23 +239,21 @@ class Client extends EventEmitter {
...
-
     return new this._Promise((resolve, reject) => {
       this._connect((error) => {
         if (error) {
           reject(error)
         } else {
           resolve()
         }
-      })
+      }).then(() => resolve())
```
[the remark this answers is not in the document]
Comment (jonathanl-bq) [reply packages/vertica-nodejs/lib/client.js:256]: @alexr-bq deemed this necessary to fix some of the failing tests, I believe. I can try to replace it and see if it affects anything.

[the remark this answers is not in the document]
Comment (alexr-bq) [reply packages/vertica-nodejs/lib/client.js:271]: I think we should actually remove this line. It fixes some unit test cases but I don't think it fixes any real cases but might cause eating of errors

File: packages/vertica-nodejs/lib/client.js [hunk 2]
```diff
@@ -132,24 +133,24 @@ class Client extends EventEmitter {
...
     if (addresses.length > 0) {
       // There are resolved addresses we haven't tried yet, so try the next address
-      this._connectToNextAddress(nodes, addresses)
+      await this._connectToNextAddress(nodes, addresses)
     } else if (nodes.length > 0) {
       // There are no more resolved addresses for the current node, so resolve the host for the next node
       var node = nodes.shift()
-      this._resolveHost(node)
-        .then(((shuffled_addresses) => {
+      await this._resolveHost(node)
```
[the remark this answers is not in the document]
Comment (jonathanl-bq) [reply packages/vertica-nodejs/lib/client.js:143]: I tried merging this suggestion, but it causes several test breaks.

File: packages/vertica-nodejs/lib/client.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ var defaults = require('./defaults')
 var Connection = require('./connection')
 const { parseCommandLine } = require('typescript')
 const { reject } = require('bluebird')
+const { SSL_OP_EPHEMERAL_RSA } = require('constants')
```
Comment (jeremyp-bq) [inline packages/vertica-nodejs/lib/client.js:17]: Is this needed, I don't see it referenced?

Comment (DMickens) [reply packages/vertica-nodejs/lib/client.js:17]: +1, maybe an accidental automatic import?

File: packages/vertica-nodejs/Makefile [hunk 4]
```diff
@@ -4,7 +4,7 @@ connectionString=postgres://
 
 params := $(connectionString)
 
-node-command := xargs -n 1 -I file node file $(params)
+node-command := xargs -n 1 -I file sh -c "node file $(params) || true"
```
Comment (DMickens) [inline packages/vertica-nodejs/Makefile:7]: what does this do?

Comment (jeremyp-bq) [reply packages/vertica-nodejs/Makefile:7]: I believe xargs was stopping after the first error.  This allows it to continue and run all the tests, even if one failed.

Comment (DMickens) [reply packages/vertica-nodejs/Makefile:7]: That's a better approach. Although now that I think about it, there are a bunch of tests that build off of the previous. We'll just need to keep that in mind when analyzing test results. One early failure may or may not be the cause of later failures in that suite

Comment (williammatherjones) [reply packages/vertica-nodejs/Makefile:7]: That sounds really messy.   Does this mean that one test does setup that is used by another?

**Review by DMickens — COMMENTED** [review]
Can you give me some context for the reason behind the async changes, and maybe some idea of the user facing impact it might have, if there even is any user facing impact?

File: packages/vertica-nodejs/Makefile (hunk 4 above)
Comment (DMickens) [reply packages/vertica-nodejs/Makefile:7]: I think the right answer to that question is yes. Mocha uses a 'describe' function to give context to what I suppose would be called a test suite, but it's all within the same file. A file can have multiple test suites. And within those test suites are individual tests that pass or fail individually, but some of them are dependent on actions occurring in previous tests of the same suite

**Comment by jonathanl-bq** [comment]
@DMickens The async changes do not actually have a user facing impact as far as I can tell. The reason for those changes is because some unit tests were failing using the original code. I believe certain promises weren't being run in the correct sequence before, so this fixes that. For the record, the original code also returned a promise from client.connect(), so making client.connect() async shouldn't be a breaking change. For instance, the example apps work fine as is still.

File: packages/vertica-nodejs/lib/client.js (hunk 3 above)
Comment (jonathanl-bq) [reply packages/vertica-nodejs/lib/client.js]: Yeah I think this is accidental. Removed now.

**Review by DMickens — APPROVED** [review]
Thanks for the explanation. If there is no user facing impact to discuss I think this is good.

**Review by jeremyp-bq — APPROVED** [review]

**Pull request merged**